replica/replica.go

```diff
@@ -194,8 +189,9 @@ func (replica *replica) buildSignedBlock() block.SignedBlock {
 	// TODO: We should put more than one transaction into a block.
 	transactions := tx.Transactions{}
```

Comment:
We should probably pre-allocate the capacity for this slice to reduce multiple runtime allocations:

```go
transactions := make(tx.Transactions, 0, block.MaxTransactions)
```